### DISCUSSION TITLE
Overwrite Menyimpan Data Frame ke Dalam Specific File Excel & Specific Sheet Existing - Python

### DISCUSSION THREAD

**Discussion by ELITSZORD** [teamalgoritma/community#814] [Q&A] [answered]
Dear Algoritma team,

Jadi kita ada file excel dengan beberapa worksheet berisi data2 dan df baru. Bagaimana cara df baru yang kita buat bisa di save dalam excel file existing dimana df tersebut men overwrite df pada sheet existing tertentu? menggunakan python.

Berikut kode yang saya coba namun ada kendala:

![COD](https://github.com/teamalgoritma/community/assets/114390619/b9e274e9-b8f6-4288-802c-dc4aae7b7910)

Error: 
![COD1](https://github.com/teamalgoritma/community/assets/114390619/709c3c15-4ec3-4cb4-b93b-013036ba9793)
![COD2](https://github.com/teamalgoritma/community/assets/114390619/a24b3da7-10d6-457a-9de0-b355c49ebad0)

Selain itu file ketika mau di tutup harus save as dan tipe data jadi berubah pada file excelnya.

Mohon bantuannya temen2. 

Terimakasih

**Comment by jafarhf1** [comment]
Dear Pak @ELITSZORD ,

Mengenai pertanyaannya, untuk dapat melakukan saving atau overwrite file excel bisa langsung menggunakan kode atau method `to_excel` saja pak. Contoh misalnya Bapak punya dataframe `df1` kemudian ingin disimpan ke file excel dengan naman `data.xlsx`  dengan sheet `Sheet1`. Maka kodenya adalah `df1.to_excel("data.xlsx",  sheet_name = "Sheet1")`. Pastikan file excelnya ditutup terlebih dahulu dan tidak sedang dibuka. Semoga membantu ya Pak

Best regards,
Team Algoritma

**Reply by ELITSZORD** [reply]
@jafarhf1 Gak bisa mas pake itu. Jadi begini kan saya ada file excel dengan nama XX.xslx dimana ada beberapa worksheet kemudian saya wrangling dan mendapatkan df baru, dimana df baru ini mau saya tiban/overwrite di sheet A pada file yang sama yaitu XX.xslx. dan pengennya sheet yang lain selain sheet A tetep ada dan mempertahankan tipe datanya..

**Reply by jafarhf1** [reply, marked as answer]
Dear Pak @ELITSZORD ,

Baik pak apakah sudah dicoba dengan kode berikut?

```
with pd.ExcelWriter("XX.xlsx", engine="openpyxl", mode='a', if_sheet_exists = 'replace') as writer:
    df_baru.to_excel(writer, sheet_name="sheet A")
```
Seharusnya sheet-sheet yang selain sheet A masih tetap ada pak mungkin bisa dicoba terlebih dahulu. Terima kasih

Best regards,
Team Algoritma
